problem_builder/mixins.py

```diff
@@ -0,0 +1,123 @@
+from lazy import lazy
+from xblock.fields import String, Boolean, Scope
+from xblockutils.helpers import child_isinstance
+from xblockutils.resources import ResourceLoader
+
+
+loader = ResourceLoader(__name__)
+
+
+# Make '_' a no-op so we can scrape strings
+def _(text):
+    return text
+
+
+def _normalize_id(key):
+    """
+    Helper method to normalize a key to avoid issues where some keys have version/branch and others don't.
+    e.g. self.scope_ids.usage_id != self.runtime.get_block(self.scope_ids.usage_id).scope_ids.usage_id
+    """
+    if hasattr(key, "for_branch"):
+        key = key.for_branch(None)
+    if hasattr(key, "for_version"):
+        key = key.for_version(None)
+    return key
+
+
+class XBlockWithTranslationServiceMixin(object):
+    """
+    Mixin providing access to i18n service
+    """
+    def _(self, text):
+        """ Translate text """
+        return self.runtime.service(self, "i18n").ugettext(text)
+
+
+class EnumerableChildMixin(XBlockWithTranslationServiceMixin):
+    CAPTION = _(u"Child")
+
+    show_title = Boolean(
+        display_name=_("Show title"),
+        help=_("Display the title?"),
+        default=True,
+        scope=Scope.content
+    )
+
+    @lazy
+    def siblings(self):
+        return self.get_parent().steps
```

Comment:
@itsjeyd mixin should not contain `siblings` - classes that use this mixin should provide their own implementations, as their siblings are different (i.e. steps should do this, while questions should return `self.get_parent().questions`, or something).